File: AndroidStudioProjects/Sample/app/src/main/java/com/example/sample/SecondActivity.java
package com.example.sample;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {

    private TextView text;
    private Button but;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        but = findViewById(R.id.button2);
        text =findViewById(R.id.textView2);

        String name = getIntent().getStringExtra("Message");

        text.setText(name);

        but.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String str = "This is Second Activity";
                Intent i = new Intent();
                i.putExtra("MS",str);
                setResult(RESULT_OK,i);
                finish();

            }
        });
    }
}

File: AndroidStudioProjects/ActivityDemo/app/src/main/java/com/example/activitydemo/SecondActivity.kt
package com.example.activitydemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class SecondActivity : AppCompatActivity() {

    private val TAG = "Life cycle"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        Log.i(TAG,"onCreate called")
    }

    override fun onStart() {
        super.onStart()
        Log.i(TAG,"onStart called")
    }

    override fun onResume() {
        super.onResume()
        Log.i(TAG,"onResume called")
    }

    override fun onPause() {
        super.onPause()
        Log.i(TAG,"onPause called")
    }

    override fun onStop() {
        super.onStop()
        Log.i(TAG,"onStop called")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i(TAG,"onDestroy called")
    }
}